pkg/component/plutono/plutono.go

```diff
@@ -84,6 +161,546 @@ func (p *plutono) WaitCleanup(ctx context.Context) error {
+		deployment.Labels = utils.MergeStringMaps(deployment.Labels, map[string]string{v1beta1constants.LabelRole: v1beta1constants.LabelMonitoring})
+		deployment.Spec.Template.Spec.AutomountServiceAccountToken = pointer.Bool(false)
+		deployment.Spec.Template.Spec.PriorityClassName = v1beta1constants.PriorityClassNameSeedSystem600
+		deployment.Spec.Template.Labels = utils.MergeStringMaps(deployment.Spec.Template.Labels, map[string]string{
+			v1beta1constants.LabelRole:                                         v1beta1constants.LabelMonitoring,
+			"networking.gardener.cloud/to-seed-prometheus":                     v1beta1constants.LabelNetworkPolicyAllowed,
+			gardenerutils.NetworkPolicyLabel("aggregate-prometheus-web", 9090): v1beta1constants.LabelNetworkPolicyAllowed,
+			gardenerutils.NetworkPolicyLabel("seed-prometheus-web", 9090):      v1beta1constants.LabelNetworkPolicyAllowed,
+		})
+		deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env, []corev1.EnvVar{
+			{Name: "PL_AUTH_BASIC_ENABLED", Value: "true"},
+			{Name: "PL_AUTH_DISABLE_LOGIN_FORM", Value: "false"},
+		}...)
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+			Name:      "plutono-storage",
+			MountPath: "/var/plutono-storage",
+		})
+		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: "plutono-storage",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		})
+	} else {
+		deployment.Labels = utils.MergeStringMaps(deployment.Labels, map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleMonitoring})
+		deployment.Spec.Template.Spec.PriorityClassName = v1beta1constants.PriorityClassNameShootControlPlane100
+		deployment.Spec.Template.Labels = utils.MergeStringMaps(deployment.Spec.Template.Labels, map[string]string{
+			v1beta1constants.GardenRole:                              v1beta1constants.GardenRoleMonitoring,
+			gardenerutils.NetworkPolicyLabel("prometheus-web", 9090): v1beta1constants.LabelNetworkPolicyAllowed,
+		})
+		deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env, []corev1.EnvVar{
+			{Name: "PL_AUTH_BASIC_ENABLED", Value: "false"},
+			{Name: "PL_AUTH_DISABLE_LOGIN_FORM", Value: "true"},
+			{Name: "PL_AUTH_DISABLE_SIGNOUT_MENU", Value: "true"},
+		}...)
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+			Name:      "plutono-storage",
+			MountPath: "/var/lib/plutono",
+		})
+		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: "plutono-storage",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{
+					SizeLimit: &sizeLimit,
```

Comment:
@istvanballok @rickardsjp Why do we have this size limit only for the plutono of shoots and not for the one for seeds?

Reply:
> Why do we have this size limit only for the plutono of shoots and not for the one for seeds?

The size limit was added to the Grafana instance that the shoot owner can access in this commit:

https://github.com/gardener/gardener/pull/7007/commits/1f055ff5400dd08bd603faead89ffd0a84d417a2 of https://github.com/gardener/gardener/pull/7007

We could add a size limit similarly to the (now) Plutono instance in the Garden namespace.

The motivation for adding a size limit was that although Grafana/Plutono is configured so that new dashboards can not be added by end users, if there was a way (not known to me) for shoot owners to let Plutono create files that could fill up the disk, then we could prevent that from happening. The garden namespace Plutono is only accessible to the more trusted Gardener operators (and not to shoot owners), but we could add a size limit there as well, for consistency. Thank you for pointing it out!

Reply:
Adapted the change in 7a8f795a9aee272c03e2b1430e9f7e76bfd07c93 PTAL.